### PR TITLE
Fixes #7158 - Allows adding of hex color to the theme-color

### LINKED ISSUE #7158: Add hex color truncates pasted color
- Open up theming setting as admin
- paste hex color, e.g. #99CC00
- see that it is truncated to #99CC0 as only 6 digits are allowed

--> Either allow both for input #99CC00 and 99CC00
--> Or delete "#", so that pasting #99CC00 is 99CC00

### PR DESCRIPTION
**Pull request by Abijeet**
Greetings,

Please find the merge request that fixes #7158

I've done the following,

1. Increased max length to allow for 7 characters.
2. Checking and adding # if not already present on the `change` event.
3. Added `jscolor {hash:true}` class so that user does not enter a 7 character colour by mistake.

---
Abijeet

PS : Sorry for the extra commits, the code indentation setting on Netbeans was a little different for me.

### CODE REVIEW

**Comment by Abijeet** [comment]
Will check the failed cases.

**Comment by Abijeet** [comment]
I've fixed the test cases, now it's failing because I forgot to sign off a few commits. I hope that's not an issue for now. I'll be careful in the future.

**Comment by MorrisJobke** [comment]
cc @nextcloud/theming @nextcloud/javascript

File: apps/theming/templates/settings-admin.php [hunk 1]
```diff
@@ -62,7 +62,7 @@
 	<div>
 		<label>
 			<span><?php p($l->t('Color')) ?></span>
-			<input id="theming-color" type="text" class="jscolor" maxlength="6" value="<?php p($_['color']) ?>" />
+			<input id="theming-color" type="text" class="jscolor {hash:true}" maxlength="7" value="<?php p($_['color']) ?>" />
```
Comment (juliushaertl) [inline apps/theming/templates/settings-admin.php]: I'd prefer using the data-jscolor attribute here.

**Review by juliusknorr — CHANGES_REQUESTED** [review]
Thanks for your PR @Abijeet 
Works great, only a small nitpick, then it's good from my side 😉

**Comment by Abijeet** [comment]
@juliushaertl - Done! :)

**Comment by MorrisJobke** [comment]
@Abijeet I fixed the commits and resubmitted your PR in #7426.

Thanks for your contribution 👍  I will close this PR here and we merge the other one :)

**Pull request closed without merging**